umich_api/api_utils.py

```diff
@@ -47,31 +51,39 @@ def __init__(self, base_url, client_id, client_secret, client_scope, api_json =
         with open(api_json, encoding='utf-8') as api_file:
             apis = json.loads(api_file.read())
 
-        if client_scope in apis: 
-            api = apis.get(client_scope)
-            self.token_url = api.get('token_url')
-            self.api_call = sleep_and_retry(limits(calls=api.get('limits_calls'), period=api.get('limits_period'))(self._api_call))
-            self.access_token = self.get_access_token(self.token_url)
-        else: 
-            raise Exception(f"Scope {client_scope} not in known API dict")
+        self.scopes = defaultdict(dict)
+        # Setup all of the calls to the apis with the limits
+        for (client_scope, api) in viewitems(apis): 
+            self.scopes[client_scope]["api_call"] = sleep_and_retry(limits(calls=api.get('limits_calls'), period=api.get('limits_period'))(self._api_call))
+            # Each client_scope seems to need it's own access token. Just get them all here first
+            if "access_token" not in self.scopes[client_scope]:
```

Comment:
How about renewing expired access_token? Is the renewal process already been handled somewhere?

Reply:
I created #12 to handle this renewa.l.

Reply:
one idea for the token renewal process to work. I see the API call is a 2 step process 1. instantiating the API obj that holds all the key, secret, scope 2. the call the API(GET, POST..) with the newly created object. 
I see making it as one call, when instantiating the API call you to give the token_url and Content_url and the utils know how to get the token and make the content call. This way when a new apil_util obj is instantiated we will always ensure that content call will return the desired result.

you might need to give a bunch of thining when making an object like 1. scope 2, token_url 3. content_url  4. rate limit etc.

Reply:
Can you add these comments to #12 and I (or whomever works on the renewal) will look at them there? Thanks!